### PR TITLE
Handling multiple form sections with FormPartial component

### PR DESCRIPTION
**Pull request by erhangundogan**
Previous multiple form sections PR was having some problems:
- Auto submit bug
- Complex submit mechanism
- Bloated `FormBuilder` component

Therefore I have re-worked on the feature and changed the way how multiple forms handled.
**The idea is simple: I am passing getter/setter methods down to the component instead of complex internal getter/setter. And also it's using HTML form element to handle form submission naturally rather than artificial submit operation.**

Please see the key notes below:
- New `FormPartial` component is the striped out version of `FormBuilder`
- Removed unnecessary `FormContext` component and it's extensions from `FormBuilder`
- Check out `/examples/PartialForms/index.tsx` for usage.
- Yet another [validation library](https://github.com/validatorjs/validator.js) usage `/examples/PartialForms/validate.ts`
- Tests will be added if all goes well
- See the usage below with gifs

https://recordit.co/lLEFntNzjf

https://recordit.co/iwz76EQBwO

### CODE REVIEW

File: examples/PartialForms/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  }, [formValues]);
+
+  const onSubmit = (event: FormEvent) =>{
+    event.preventDefault();
+    const errors = validate(formValues);
+    const isValidForm = Object.keys(errors).length === 0;
+
+    if (isValidForm) {
+      console.log(formValues);
```
Comment (peeyushsingla) [inline examples/PartialForms/index.tsx]: extra `console.log` statement.

File: examples/PartialForms/index.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  const getValue = (field: string) => formValues[field];
+  const setValue = (field: string) => (value: any) => setFormValues({ ...formValues, [field]: value });
+
+  return (
+    <div className="form-container">
+      <div className="pure-form pure-form-stacked">
+        <form onSubmit={ onSubmit } noValidate={ true }>
+          <PersonalDetailsForm { ...{ getValue, setValue, validationErrors } } />
+          <ContactDetailsForm { ...{ getValue, setValue, validationErrors } } />
```
Comment (maniodev) [inline examples/PartialForms/index.tsx]: Could you please extend the example by passing some fields values ?

File: examples/PartialForms/index.tsx (hunk 1 above)
Comment (erhangundogan) [reply examples/PartialForms/index.tsx]: it's not an extra. it's for an example output 😄

File: examples/PartialForms/index.tsx (hunk 2 above)
Comment (erhangundogan) [reply examples/PartialForms/index.tsx]: addressed here https://github.com/Savedo/re-form/pull/33/commits/9f65d5de4b5acaf5accb000183109d95c164ca42

**Pull request merged**